.github/workflows/cli.yml

```diff
@@ -31,55 +42,428 @@ jobs:
 
   notify:
-    if: failure()
     runs-on: ubuntu-latest
-    needs: apic-extension-live-test
     environment: engineering
+    needs: [setup, test]
+    if: (github.event_name == 'schedule' || github.event_name == 'workflow_dispatch') && !cancelled()
     steps:
+      - name: Download all test result artifacts
+        uses: actions/download-artifact@v4
+        with:
+          path: test-results/
+          pattern: test-results-*
+          merge-multiple: true
+
       - name: Generate Email Content
         id: generate-email-content
         run: |
-          echo "to=${{ secrets.AUTHOR_MAIL_ADDRESSES }}" >> $GITHUB_ENV
-          echo "body=Workflow URL: ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}" >> $GITHUB_ENV
-          echo "subject=[APICENTER] Weekly APIC Extension Live Test Failed" >> $GITHUB_ENV
+          test_cases='${{ needs.setup.outputs.test-cases }}'
+          python_versions='${{ needs.setup.outputs.python-versions }}'
+          python_versions_array=$(echo "$python_versions" | jq -r '.[]')
+          python_versions_count=$(echo "$python_versions" | jq '. | length')
+
+          # Initialize counters
+          total_cases=0
+          total_success=0
+          total_failed=0
+          failed_cases=""
+
+          if [ "$test_cases" = '["full-suite"]' ]; then
+            cases_per_version="TBD"
+          else
+            cases_per_version="TBD"
+          fi
+
+          email_body="<h2>📊 CLI Weekly Test Results Summary</h2>
+          <p><strong>Test Execution Date:</strong> $(date '+%Y-%m-%d %H:%M:%S UTC')<br>
+          <strong>Repository:</strong> azure-cli-extensions (apic-extension)<br>
+          <strong>Test Cases per Python Version:</strong> $cases_per_version<br>
+          <strong>Python Versions Tested:</strong> $(echo "$python_versions" | jq -r 'join(", ")')</p>
+
+          <h3>Overall Summary:</h3>
+          <table border='1' cellpadding='8' cellspacing='0' style='border-collapse: collapse; font-family: monospace;'>
+            <thead style='background-color: #f0f0f0;'>
+              <tr>
+                <th style='text-align: left; padding: 8px;'>Python Version</th>
+                <th style='text-align: center; padding: 8px;'>Total Cases</th>
+                <th style='text-align: center; padding: 8px;'>Successful</th>
+                <th style='text-align: center; padding: 8px;'>Failed</th>
+                <th style='text-align: center; padding: 8px;'>Success Rate</th>
+              </tr>
+            </thead>
+            <tbody>"
+          # Process results for each Python version
+          test_job_result='${{ needs.test.result }}'
+          while IFS= read -r version; do
+            # Find result files for this Python version
+            result_files=""
+            for pattern in "result-${version}-*.json" "result-*${version}*.json" "*python-${version}*.json" "*${version}*.json"; do
```

Comment:
Is the file name specified by ourselves or generated by a tool? Are we able to know what's the exact file name pattern? It better to generate report with a single file name pattern instead of multiple file name patterns.